pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java

```diff
@@ -92,52 +91,34 @@ AggregationInfo buildAggregationInfo() {
-        }
-        if (children.size() == 1) {
-          // single column DISTINCT query
-          // e.g SELECT DISTINCT(col) FROM foo
-          if (children.get(0) instanceof StarExpressionAstNode) {
-            // DISTINCT(*) is not supported yet. User has to specify each column that should participate in DISTINCT
-            throw new Pql2CompilationException(
-                "Syntax error: Pinot currently does not support DISTINCT with *. Please specify each column name as argument to DISTINCT function");
-          }
-          expression = TransformExpressionTree.getStandardExpression(children.get(0));
-        } else {
-          // multi-column DISTINCT query
-          // e.g SELECT DISTINCT(col1, col2) FROM foo
-          // we will pass down the column expression to execution code
-          // as col1:col2
-          Set<String> expressions = new HashSet<>();
-          StringBuilder distinctColumnExpr = new StringBuilder();
-          int numChildren = children.size();
-          for (int i = 0; i < numChildren; ++i) {
-            expression = TransformExpressionTree.getStandardExpression(children.get(i));
-            if (expressions.add(expression)) {
-              // deduplicate the columns
-              if (i != 0) {
-                distinctColumnExpr.append(DISTINCT_MULTI_COLUMN_SEPARATOR);
-              }
-              distinctColumnExpr.append(expression);
-            }
+
+      // TODO: Remove all distinct special-casing.
+      // DISTINCT(*) is not supported yet. User has to specify each column that should participate in DISTINCT
+      if (_name.equalsIgnoreCase(AggregationFunctionType.DISTINCT.getName()) && children.size() == 1 && children
+          .get(0) instanceof StarExpressionAstNode) {
+        throw new Pql2CompilationException(
+            "Syntax error: Pinot currently does not support DISTINCT with *. Please specify each column name as argument to DISTINCT function");
+      }
+
+      Set<String> expressions = new HashSet<>();
+      StringBuilder distinctColumnExpr = new StringBuilder();
+      int numChildren = children.size();
+      for (int i = 0; i < numChildren; ++i) {
+        expression = TransformExpressionTree.getStandardExpression(children.get(i));
+        if (expressions.add(expression)) {
+          // deduplicate the columns
+          if (i != 0) {
+            distinctColumnExpr.append(AGGREGATION_FUNCTION_ARG_SEPARATOR);
           }
-          expression = distinctColumnExpr.toString();
+          distinctColumnExpr.append(expression);
         }
       }
+      expression = distinctColumnExpr.toString();
     }
 
     AggregationInfo aggregationInfo = new AggregationInfo();
     aggregationInfo.setAggregationType(_name);
-    aggregationInfo.putToAggregationParams(COLUMN_KEY_IN_AGGREGATION_INFO, expression);
+    aggregationInfo.putToAggregationParams(CompilerConstants.COLUMN_KEY_IN_AGGREGATION_INFO, expression);
```

Comment:
should we instead fix AggregationInfo to take in multiple arguments instead?

Reply:
Yes we should. I didn't want to break backward compatibility in this PR though.